File: app/src/main/java/com/example/dcct/model/RegisterModel.java
package com.example.dcct.model;

import com.example.dcct.bean.BackResultData;
import com.example.dcct.bean.RegisterUserEntity;

public interface RegisterModel{
    void postRegisterInfor(RegisterUserEntity registerUserEntity,ReturnRegisterResult returnRegisterResult);
    interface ReturnRegisterResult{
        void onComplete(BackResultData backResultData);
        void onError(String msg);
    }

}

File: app/src/main/java/com/example/dcct/model/RegisterModel.kt
package com.example.dcct.model

import com.example.dcct.bean.BackResultData
import com.example.dcct.bean.RegisterUserEntity

interface RegisterModel {
    fun postRegisterInfo(registerUserEntity: RegisterUserEntity, returnRegisterResult: ReturnRegisterResult)
    interface ReturnRegisterResult {
        fun onComplete(backResultData: BackResultData<*>)
        fun onError(msg: String)
    }
}